sdk/src/azure/core/az_mqtt5_rpc_client.c

```diff
@@ -13,97 +13,60 @@ AZ_NODISCARD az_mqtt5_rpc_client_options az_mqtt5_rpc_client_options_default()
   return (az_mqtt5_rpc_client_options){ .subscribe_timeout_in_seconds
                                         = AZ_MQTT5_RPC_DEFAULT_TIMEOUT_SECONDS,
                                         .publish_timeout_in_seconds
-                                        = AZ_MQTT5_RPC_DEFAULT_TIMEOUT_SECONDS };
+                                        = AZ_MQTT5_RPC_DEFAULT_TIMEOUT_SECONDS,
+                                        .subscription_topic_format = AZ_SPAN_EMPTY,
+                                        .request_topic_format = AZ_SPAN_EMPTY };
 }
 
 AZ_NODISCARD az_result az_rpc_client_get_subscription_topic(
     az_mqtt5_rpc_client* client,
     az_span out_subscription_topic,
     int32_t* out_topic_length)
 {
-  _az_PRECONDITION_VALID_SPAN(client->_internal.model_id, 1, false);
-  _az_PRECONDITION_VALID_SPAN(client->_internal.client_id, 1, false);
-  _az_PRECONDITION_VALID_SPAN(client->_internal.command_name, 1, false);
-  *out_topic_length = az_span_size(client->_internal.model_id)
-      + az_span_size(client->_internal.client_id) + az_span_size(client->_internal.command_name)
-      + 29;
-  if (!_az_span_is_valid(out_subscription_topic, *out_topic_length, true))
-  {
-    return AZ_ERROR_OUT_OF_MEMORY;
-  }
-
-  // TODO: Create generic function to create topics from format string
-  out_subscription_topic = az_span_copy(out_subscription_topic, AZ_SPAN_FROM_STR("vehicles/"));
-  out_subscription_topic = az_span_copy(out_subscription_topic, client->_internal.model_id);
-  out_subscription_topic = az_span_copy(out_subscription_topic, AZ_SPAN_FROM_STR("/commands/+/"));
-  out_subscription_topic = az_span_copy(out_subscription_topic, client->_internal.command_name);
-  out_subscription_topic = az_span_copy(out_subscription_topic, AZ_SPAN_FROM_STR("/__for_"));
-  out_subscription_topic = az_span_copy(out_subscription_topic, client->_internal.client_id);
-  out_subscription_topic = az_span_copy_u8(out_subscription_topic, '\0');
-
-  return AZ_OK;
+  return az_rpc_get_topic_from_format(
+      client->_internal.options.subscription_topic_format,
+      client->_internal.model_id,
+      AZ_SPAN_FROM_STR("+"),
+      client->_internal.client_id,
+      _az_span_is_valid(client->_internal.command_name, 1, 0) ? client->_internal.command_name
+                                                              : AZ_SPAN_FROM_STR("+"),
+      out_subscription_topic,
+      out_topic_length);
 }
 
+// Replaces + in subscription topic with server client id
 AZ_NODISCARD az_result az_rpc_client_get_response_topic(
     az_mqtt5_rpc_client* client,
     az_span server_client_id,
```

Comment:
Can we instead pass an `az_mqtt5_rpc_server_codec` object? We shouldn't be leaking internals within the codec.

Reply:
This is the code server object - this will be clearer after the rename